grpc.go

```diff
@@ -768,10 +770,16 @@ func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest)
 		return &pb.WaitProcessResponse{}, err
 	}
 
+	proc.exitCodeCh.RLock()
+
 	defer func() {
+		proc.exitCodeCh.RLock()
```

Comment:
No need for the extra func() closure. Please just use `defer proc.exitCodeCh.RLock()`.

Reply:
Got it.

Fupan